holoviews/core/overlay.py

```diff
@@ -136,7 +135,15 @@ def __add__(self, other):
 
 
     def __mul__(self, other):
-        if not isinstance(other, ViewableElement):
+        if type(other).__name__ == 'DynamicMap':
+            from .spaces import OverlayCallable
+            def dynamic_mul(*args, **kwargs):
+                element = other[args]
+                return self * element
+            callback = OverlayCallable(dynamic_mul, inputs=[self, other])
+            return other.clone(shared_data=False, callback=callback,
+                               streams=[])
+        elif not isinstance(other, ViewableElement):
```

Comment:
We really need to refactor ``__mul__`` (dynamic and non-dynamic) into an operation for 2.0. It was bad enough before considering dynamic and now we have `` _dynamic_mul`` methods and ``dynamic_mul`` inner functions all over the place.